components/server/src/api/verification-service-api.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { HandlerContext, ServiceImpl } from "@connectrpc/connect";
+import { ApplicationError, ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+import { VerificationService as VerificationServiceInterface } from "@gitpod/public-api/lib/gitpod/v1/verification_connect";
+import {
+    SendPhoneNumberVerificationTokenRequest,
+    SendPhoneNumberVerificationTokenResponse,
+    VerifyPhoneNumberVerificationTokenRequest,
+    VerifyPhoneNumberVerificationTokenResponse,
+} from "@gitpod/public-api/lib/gitpod/v1/verification_pb";
+import { inject, injectable } from "inversify";
+import { VerificationService } from "../auth/verification-service";
+import { getExperimentsClientForBackend } from "@gitpod/gitpod-protocol/lib/experiments/configcat-server";
+import { ctxUserId } from "../util/request-context";
+import { UserService } from "../user/user-service";
+import { formatPhoneNumber } from "../user/phone-numbers";
+import { validate as uuidValidate } from "uuid";
+
+@injectable()
+export class VerificationServiceAPI implements ServiceImpl<typeof VerificationServiceInterface> {
+    @inject(VerificationService) private readonly verificationService: VerificationService;
+    @inject(UserService) private readonly userService: UserService;
+
+    async sendPhoneNumberVerificationToken(
+        req: SendPhoneNumberVerificationTokenRequest,
+        _: HandlerContext,
+    ): Promise<SendPhoneNumberVerificationTokenResponse> {
+        if (!req.phoneNumber) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "phoneNumber is required");
+        }
+
+        const userId = ctxUserId();
+        const user = await this.userService.findUserById(userId, userId);
+
+        // Check if verify via call is enabled
+        const phoneVerificationByCall = await getExperimentsClientForBackend().getValueAsync(
+            "phoneVerificationByCall",
+            false,
+            {
+                user,
+            },
+        );
+        const channel = phoneVerificationByCall ? "call" : "sms";
+        const { verificationId } = await this.verificationService.sendVerificationToken(
```

Comment:
we are missing analytics enent here which is present in JSON-RPC

I think much of code from JSON-RPC should have been moved to VerificationService and not copied in API layer, here we shoudl not have logic but rather only translations